### DISCUSSION TITLE
Is it possible to share a user-implemented mapping method between mappers ?

### DISCUSSION THREAD

**Discussion by elendil-software** [riok/mapperly#1430] [Q&A] [answered]
Hi,

I'm in the same situation than this issue : #1107 

Like in this issue I need to map gRPC timestamp.

So I proceeded the same way explained in the issue : 

```cs
[Mapper]
public static partial class Mapper
{
    public static partial GrpcResponse MapToGrpcResponse(this Response response);

    private static Timestamp DateTimeOffsetToTimestamp(DateTimeOffset source)
    {        
        return Timestamp.FromDateTimeOffset(source);
    }
}
```
But I don't have only one mapper that need this conversion. Is it possible to share the `DateTimeOffsetToTimestamp` method between several mappers ?

Regards,

**Comment by sveinhelge** [comment, marked as answer]
Are you looking for external mappings, [see Mapperly docs](https://mapperly.riok.app/docs/configuration/user-implemented-methods/#use-external-mappings)

```csharp
[Mapper]
[UseStaticMapper(typeof(BananaMapper))]
public static partial class BoxMapper
{
  public static partial BananaBox MapBananaBox(BananaBoxDto dto);
}

public static class BananaMapper
{
  public static Banana MapBanana(BananaDto dto)
    => new Banana(dto.Weigth);
}
```

**Reply by latonz** [reply]
The usage of external mappers is one option, anither one is using a common base class for the mappers.

**Reply by elendil-software** [reply]
> The usage of external mappers is one option, anither one is using a common base class for the mappers.

I thought to the second option. But it would not be possible with extensions method as they requires the class to be static
